File: backend/src/main/java/com/hospedaya/backend/application/service/base/RecomendacionService.java
package com.hospedaya.backend.application.service.base;

import com.hospedaya.backend.domain.entity.Alojamiento;

import java.util.List;

public interface RecomendacionService {
    List<Alojamiento> recomendarPorUsuario(Long usuarioId, int limit);
}

File: hospedaya-frontend/src/app/services/recomendacion.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { AlojamientoResponseDTO } from './alojamiento.service';

@Injectable({ providedIn: 'root' })
export class RecomendacionService {
  private baseUrl = 'https://hospedaya-proyecto.onrender.com/recomendaciones';

  constructor(private http: HttpClient) {}

  porUsuario(usuarioId: number, limit = 8): Observable<AlojamientoResponseDTO[]> {
    return this.http.get<AlojamientoResponseDTO[]>(`${this.baseUrl}/usuario/${usuarioId}?limit=${limit}`);
  }
}
